_studio/shared/umc/codec/jpeg_dec/src/jpegdec_base.cpp

```diff
@@ -157,18 +159,22 @@ JERRCODE CJPEGDecoderBase::Clean(void)
 
 #define BS_BUFLEN 16384
 
-JERRCODE CJPEGDecoderBase::SetSource(
-  CBaseStreamInput* pInStream)
+JERRCODE CJPEGDecoderBase::SetSource(const uint8_t* pBuf, size_t buflen)
 {
-  JERRCODE jerr;
+  JERRCODE jerr = m_stream_in.Open(pBuf, buflen);
 
-  jerr = m_BitStreamIn.Attach(pInStream);
   if(JPEG_OK != jerr)
     return jerr;
 
+  m_BitStreamIn.Attach(&m_stream_in);
```

Comment:
you forgot to check attach return status

Reply:
It not make sense, Attach() can return only  JPEG_OK

JERRCODE CBitStreamInput::Attach(CBaseStreamInput* in)
{
  Detach();

  m_in = in;

  return JPEG_OK;
} // CBitStreamInput::Attach()